mp/src/game/client/momentum/ui/leaderboards/LeaderboardsTimes.cpp

```diff
@@ -529,21 +557,20 @@ void CLeaderboardsTimes::OnlineTimesVectorToLeaderboards(TimeType_t type)
     }
 }
 
-bool CLeaderboardsTimes::GetPlayerTimes(KeyValues* outPlayerInfo, bool fullUpdate)
+bool CLeaderboardsTimes::FillPlayerTimes(bool fullUpdate)
 {
-    if (!outPlayerInfo)
-        return false;
+    if (!m_bTimesNeedUpdate[TIMES_LOCAL])
+    {
+        float lastUp = gpGlobals->curtime - m_flTimesLastUpdate[TIMES_LOCAL];
+        m_bTimesNeedUpdate[TIMES_LOCAL] = fullUpdate && lastUp >= UPDATE_INTERVAL;
+    }
 
-    KeyValues *pLeaderboards = new KeyValues("leaderboards");
+    LoadLocalTimes();
 
-    // Fill local times:
-    KeyValues *pLocal = new KeyValues("local");
-    LoadLocalTimes(pLocal);
-    pLeaderboards->AddSubKey(pLocal);
 
     if (!g_pGameModeSystem->GameModeIs(GAMEMODE_UNKNOWN))
     {
-        for (int i = 1; i < TIMES_COUNT; i++) // Skip over local
```

Comment:
This comment was okay because it explained why the loop starts at 1

Reply:
Oops, I originally changed how that worked but I forgot to re-add the comment!